airflow/cli/commands/task_command.py

```diff
@@ -177,7 +176,7 @@ def _get_ti_db_access(
 
     if not exec_date_or_run_id and not create_if_necessary:
         raise ValueError("Must provide `exec_date_or_run_id` if not `create_if_necessary`.")
-    if needs_expansion(task):
+    if task.get_needs_expansion():
         if map_index < 0:
             raise RuntimeError("No map_index passed to mapped task")
```

Comment:
```suggestion
    if task.get_needs_expansion() and map_index < 0:
        raise RuntimeError("No map_index passed to mapped task")
```

Reply:
look below, there's an `elif` ... in any case, it's unrelated so i'll leave it

Reply:
yes :) I did see the elif below earlier. However, we can leave it as it is since unrelated, agree.

But I think logically 
```
if a==1:
    if b==2:
        print("Hello, ab!")
elif c==3:
    print("Hello, World!")
```

is same as
```
if a == 1 and b == 2:
    print("Hello, ab!")
elif c == 3:
    print("Hello, World!")
```
It's just that the latter has less nested branching for print("Hello, ab")

Reply:
ah yeah true i thought the elif was inside

in any case yeah i'd leave it alone because although it's true, they are functionally equivalent, it would unnecessarily distort the git history without much real benefit IMHO.